Sources/Amplitude/Plugins/iOS/UIKitUserInteractions.swift

```diff
@@ -178,40 +178,45 @@ extension NSObject {
 
 protocol ActionTrackable {
     var amp_title: String? { get }
-    func amp_shouldTrack(_ action: Selector, for event: UIEvent?) -> Bool
+    func amp_shouldTrack(_ action: Selector, for target: Any?) -> Bool
 }
 
 extension UIView: ActionTrackable {
     @objc var amp_title: String? { nil }
-    @objc func amp_shouldTrack(_ action: Selector, for event: UIEvent?) -> Bool { true }
+    @objc func amp_shouldTrack(_ action: Selector, for target: Any?) -> Bool { false }
 }
 
 extension UIButton {
     override var amp_title: String? { currentTitle }
+    override func amp_shouldTrack(_ action: Selector, for target: Any?) -> Bool {
+        actions(forTarget: target, forControlEvent: .touchUpInside)?.contains(action.description) ?? false
```

Comment:
I think this still has a couple of edge cases where actions with the same name registered to multiple control events could lead to tracking of unwanted events that we'd end up classifying incorrectly.

Is there any way we can just get the control event and use that as the filter?

Reply:
It seems vary unlikely to be able to get the control event from the parameters of `UIApplication.sendAction`. So, I don't think there is a way to know what control event the action method was associated with in case the same method is fired for multiple control events.

Reply:
Is it possible to swizzle https://developer.apple.com/documentation/uikit/uicontrol/1618211-sendactions instead? Are there any cases that are handed here that aren't sent via `sendActions`?

Reply:
I've tried swizzling this method before - apparently this method is not called by the system and meant to be called manually.

Reply:
The only way I could think of to mitigate this edge case is to roll up all the consecutive events, the coalescing approach, so that we send a single event to Amplitude for a sequence of action methods that were fired for a single interaction, no matter if the method was called by multiple control events.

Reply:
Ah, you'd need to override `[UIControl _sendActionsForEvents:withEvent:]` which is internal. You could also just attach listeners for relevant events to every control, either by crawling the view hierarchy or swizzling `addTarget`, but I don't think that is a direction we currently want to go down. Is there anything interesting in the UIEvent that gets sent along? Especially for the `valueChanged` controls.

Reply:
`UIEvent` is mostly `nil` when `valueChanged` is fired by some controls. Checking if a touch event of `UIEvent.allTouches` is ended on that view can tell us when a button is touched up though.

Reply:
OK, I think this is reasonable given the constraints.